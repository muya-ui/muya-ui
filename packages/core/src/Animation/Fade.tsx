import { CSSProperties } from 'react';
import { TransitionStatus } from 'react-transition-group/Transition';

import { IAnimationBaseParams } from './types';
import { withAnimation } from './withAnimation';
import reflow from '../utils/reflow';

function makeStyles(state: TransitionStatus, props: IAnimationBaseParams): CSSProperties {
  const {
    theme: {
      transition: { pattern: transitionPattern },
    },
    timeout,
    easing,
  } = props;
  const defaultStyle: CSSProperties = {
    opacity: 0,
    transitionProperty: 'opacity',
    transitionTimingFunction: easing || transitionPattern.easing.status,
    transitionDuration: `${timeout}ms`,
  };
  const transitionStyles: Partial<Record<TransitionStatus, CSSProperties>> = {
    entering: {
      transitionTimingFunction: easing || transitionPattern.easing.enter,
      opacity: 1,
    },
    entered: {
      opacity: 1,
    },
    exiting: {
      transitionTimingFunction: easing || transitionPattern.easing.leave,
      opacity: 0,
    },
    exited: {
      opacity: 0,
      visibility: 'hidden',
    },
  };
  return {
    ...defaultStyle,
    ...transitionStyles[state],
  };
}

const beforeEnter = (node: HTMLElement) => {
  reflow(node);
};

export default withAnimation(makeStyles, {
  beforeEnter,
});
