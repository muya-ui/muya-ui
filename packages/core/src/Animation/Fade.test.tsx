import React from 'react';
import renderer from 'react-test-renderer';

import Fade from './Fade';

test('should render correctly', () => {
  const tree = renderer
    .create(
      <Fade in={true}>
        <img
          src="https://qhstaticssl.kujiale.com/newt/29/image/png/1564467685432/EB6BD3A294454A98A65AF964C7666D94.png"
          style={{
            width: '80%',
            height: '80%',
          }}
        />
      </Fade>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
