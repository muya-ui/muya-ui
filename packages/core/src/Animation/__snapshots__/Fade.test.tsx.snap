// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<img
  src="https://qhstaticssl.kujiale.com/newt/29/image/png/1564467685432/EB6BD3A294454A98A65AF964C7666D94.png"
  style={
    Object {
      "height": "80%",
      "opacity": 1,
      "transitionDuration": "300ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "cubic-bezier(0.0, 0, 0.2, 1)",
      "width": "80%",
    }
  }
/>
`;
